agent/local/state.go

```diff
@@ -488,6 +488,9 @@ func (l *State) addCheckLocked(check *structs.HealthCheck, token string) error {
 		return fmt.Errorf("Check %q refers to non-existent service %q", check.CheckID, check.ServiceID)
 	}
 
+	// If this is a check for an aliased service, then notify the waiters.
+	l.notifyIfAliased(check.ServiceID)
```

Comment:
Wont the call to `setCheckStateLocked` below also call this. Is it necessary to call it in both places?

Reply:
Nope, it's not necessary. Just removed this one.